pkg/steps/multi_stage/gen.go

```diff
@@ -134,6 +134,17 @@ func (s *multiStageTestStep) generatePods(
 			commands = []string{"/bin/bash", "-c", CommandPrefix + step.Commands}
 		}
 		labels := map[string]string{base_steps.LabelMetadataStep: step.As}
+		// Propagate the DNSConfig from the MultiStageTestStep
```

Comment:
So what we are saying is that the resolver will propagate DNS from the job to the step level, and the code block below this will set the DNSConfig from the step to the pod.

What worries me a little is that in the tests for `multi_stage/gen.go`, we don't call into `resolve.go` at all. This probably means that the tests are assuming that the provided job and step fixtures have already been properly populated. Is that the case? Are there any cases we may not have considered that would result in DNSConfig not already being set in the step config?

Reply:
As far as my knowledge goes `gen.go` and `resolve.go` are being tested separately so yes, we assume the step has already been properly resolved.
I don't want to start a religious battle around unit tests here but I think it makes sense in that way. Say I write a test in `gen_test.go` and I call `resolve()`, if `resolve()` is broken my test fails but I get a misleading signal as I expect the test to fail iff any funcs from `gen.go` fail, instead of another one (from 'resolve.go` in this particular case).